contracts/MechMarketplace.sol

```diff
@@ -275,8 +272,9 @@ contract MechMarketplace is IErrorsMarketplace {
 
         // Record factory that created a mech
         mapAgentMechFactories[mech] = mechFactory;
-        // Add mech address into the global set
-        setMechs.push(mech);
```

Comment:
I would leave this in some form. For now there's no mech tracker, just events. `numMechs` then becomes meaningless.

Reply:
We now have the mapping, don't also need a set. The numMechs is now tracked explicitly as a variable